### PR TITLE
Fix GDAS group B restart archiving

### LINKED ISSUE #2722: Restartb files missing from previous cycle HPSS archive
### What is wrong?

In order to restart (warm start) an experiment using data stored on the HPSS archive, restarta files from the current cycle and restartb files from the previous cycle are needed. Currently restarta and restartb files are being written for 00z cycles and using only 00z cycle data. 

### What should have happened?

Restartb files should be populated in the HPSS archive for the previous cycle (18z), and restarta files for the current cycle (00z)

### What machines are impacted?

All or N/A, Hera

### Steps to reproduce

1. Clone and build develop workflow
2. Change `ARCH_WARMICFREQ=1` in config.base in order to archive restartb files in first 00z cycle (default of `ARCH_WARMICFREQ=4` is only saving restartb files every 4th day)
3. Run a simulation that archives data to HPSS at 18z and 00z cycles 
4. Check what restart files are present for the 18z and 00z cycles

### Additional information

Tests identifying the bug were run on HERA:
EXPDIR: /scratch1/NCEPDEV/da/Travis.J.Elless/exp/test_new_develop
COMROOT: /scratch2/NCEPDEV/stmp3/Travis.J.Elless/comrot/test_new_develop

### Do you have a proposed solution?

_No response_

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
# Description
Archives the GDAS restartb dataset at a 6-hour offset from the restarta dataset to allow cycled experiments the ability to restart from the archives.  The tabbing for the master archive templates was also added to improve readability.

Resolves #2722 

# Type of change
- Bug fix (fixes something broken)
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
- [x] Cycled test on Hera

# Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes

### CODE REVIEW

File: parm/archive/master_gdas.yaml.j2 [hunk 1]
```diff
@@ -5,99 +5,106 @@
...
-{% include "gdas_restarta.yaml.j2" %}
-{% endfilter %}
+        # The GDAS ICs (group b) are restarts and always lag increments by assim_freq
+        {% set ics_offset = (assim_freq | string + "H") | to_timedelta %}
+        {% if (current_cycle | add_to_datetime(ics_offset) - SDATE).days % ARCH_WARMICFREQ == 0 %}
+            {% if (ARCH_CYC - assim_freq) % 24 == cycle_HH | int %}
+                {% set save_warm_start_cycled = True %}
+            {% endif %}
+        {% endif %}
+    {% endif %}
```
Comment (DavidHuber-NOAA) [inline parm/archive/master_gdas.yaml.j2:69]: @TravisElless-NOAA Here are the actual changes to selecting when to archive restartb datasets.

**Comment by TravisElless-NOAA** [comment]
@DavidHuber-NOAA I tried running with this branch and the archive jobs are failing with the following errors
`} )^[[0m
^[[38;21m2024-06-29 03:50:38,689 - INFO     - archive     :   END: pygfs.task.archive._rename_cyclone_expt^[[0m
^[[38;5;39m2024-06-29 03:50:38,689 - DEBUG    - archive     :  returning: None^[[0m
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/da/Travis.J.Elless/git/pr_2735/global-workflow/scripts/exglobal_archive.py", line 63, in <module>
    main()
  File "/scratch1/NCEPDEV/da/Travis.J.Elless/git/pr_2735/global-workflow/ush/python/wxflow/logger.py", line 266, in wrapper
    retval = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/scratch1/NCEPDEV/da/Travis.J.Elless/git/pr_2735/global-workflow/scripts/exglobal_archive.py", line 50, in main
    arcdir_set, atardir_sets = archive.configure(archive_dict)
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch1/NCEPDEV/da/Travis.J.Elless/git/pr_2735/global-workflow/ush/python/wxflow/logger.py", line 266, in wrapper
    retval = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/scratch1/NCEPDEV/da/Travis.J.Elless/git/pr_2735/global-workflow/ush/python/pygfs/task/archive.py", line 127, in configure
    for dataset in parsed_sets.datasets.values():
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'values'
`
output dir: /scratch2/NCEPDEV/stmp3/Travis.J.Elless/comrot/test_new_atm_archive

**Comment by DavidHuber-NOAA** [comment]
@TravisElless-NOAA Apologies.  I failed to commit/push an update.  Would you please try out the updated branch?

**Comment by DavidHuber-NOAA** [comment]
Marking ready for review after performing a successful extended 4d EnVar test on Hera.

**Comment by TravisElless-NOAA** [comment]
I ran a 3DVAR atm test case with this branch for 2 days with `ARCH_WARMICFREQ=1` and I'm not getting restartb files to populate at all. Restarta files are present for each cycle corresponding to the `ARCH_CYC` value. Perhaps there is something still amiss with this? Below are the directories associated with my tests. 

expdir: /scratch1/NCEPDEV/da/Travis.J.Elless/exp/test_new_atm_archive
comroot: /scratch2/NCEPDEV/stmp3/Travis.J.Elless/comrot/test_new_atm_archive

Just an FYI: I'll be out on leave through 7/12 so I won't be able to contribute to this further until I return 7/15.

**Comment by DavidHuber-NOAA** [comment]
I am not sure how this worked for me previously, but I believe I have now fixed the mistake.  I am retesting with a 3DVar test presently and will add logs and file locations on HPSS when complete.

**Comment by DavidHuber-NOAA** [comment]
@TravisElless-NOAA I have corrected the issue and tested the functionality.  With `ARCH_CYC=00`, `ARCH_WARMICFREQ=1`, and `ARCH_FCSTICFREQ=1`, the test case was run from 2021122018 through 2021122406.  `restarta` tarballs were created on all 00z cycles for both `gfs` and `gdas` `RUN`s.  `restartb` tarballs were generated and pushed to HPSS for the 2021122118, 2021122218, and 2021122318 cycles.

The log files can be found here: `/scratch1/NCEPDEV/global/David.Huber/para/COMROOT/fix_archive/logs`.  The HPSS archives can be found here: `/NCEPDEV/emc-global/1year/David.Huber/HERA/scratch/fix_archive`.

**Comment by CatherineThomas-NOAA** [comment]
@DavidHuber-NOAA: Thanks for taking care of this!  @TravisElless-NOAA is on leave, so I took a look instead.  Your test looks good to me.  I also ran a hybrid ATM-only case over the weekend and found that the enkfgdas restart b's were also saved properly.  My test is located on HPSS here:
`/NCEPDEV/emc-da/1year/Catherine.Thomas/HERA/scratch/archive_restartb`

**Review by CatherineThomas-NOAA — APPROVED** [review]

**Review by aerorahul — APPROVED** [review]

**Pull request merged**